cmd/fileshare/main.go

```diff
@@ -145,25 +177,36 @@ func main() {
 
 	err = fileshareImplementation.Enable(meshnetIP)
 	if err != nil {
-		log.Fatalf("enabling libdrop: %s", err)
+		log.Printf("enabling libdrop: %s", err)
+		if errors.Is(err, libdrop.ErrLAddressAlreadyInUse) {
+			os.Exit(int(fileshare_process.CodeAddressAlreadyInUse))
+		}
+		os.Exit(int(fileshare_process.CodeFailedToEnable))
 	}
 
+	shutdownChan := make(chan struct{})
+
 	// Fileshare gRPC server init
 	fileshareServer := fileshare.NewServer(fileshareImplementation,
 		eventManager,
 		meshClient, fileshare.NewStdFilesystem("/"),
 		fileshare.StdOsInfo{},
-		transferHistoryChunkSize)
+		transferHistoryChunkSize,
+		shutdownChan)
 	grpcServer := grpc.NewServer()
 	pb.RegisterFileshareServer(grpcServer, fileshareServer)
 
-	var listenerFunction = internal.SystemDListener
-	if os.Getenv(internal.ListenPID) != strconv.Itoa(os.Getpid()) {
-		listenerFunction = internal.ManualListener(ConnURL, internal.PermUserRWX)
+	listener, err := net.Listen("unix", fileshare_process.FileshareSocket)
+
+	if err != nil {
+		log.Printf("Failed to open unix socket: %s", err)
```

Comment:
If the fileshared process crashes it will never restart because the socket is already open:
`Failed to open unix socket: listen unix /tmp/fileshare.sock: bind: address already in use`